Gems/ONNX/External/FindOnnx.cmake

```diff
@@ -0,0 +1,9 @@
+set(TARGET_WITH_NAMESPACE "Onnx")
+if (TARGET ${TARGET_WITH_NAMESPACE})
+    return()
+endif()
+
+set(Onnx_INCLUDE_DIR ${CMAKE_CURRENT_LIST_DIR}/onnxruntime-win-x64-gpu-1.9.0/include)
+set(Onnx_LIBS_DIR ${CMAKE_CURRENT_LIST_DIR}/onnxruntime-win-x64-gpu-1.9.0/lib)
+
+add_library(${TARGET_WITH_NAMESPACE} STATIC IMPORTED GLOBAL)
```

Comment:
This gem only builds on Windows and will fail to pass AR as this FindOnnx.cmake script will not work on non-Windows platforms.

The PAL_TRAITS for which system the ONNX gem is supported on needs to only be set to TRUE on Windows and False for other platforms
```
set(PAL_TRAIT_ONNX_SUPPORTED TRUE)
set(PAL_TRAIT_ONNX_TEST_SUPPORTED TRUE)
set(PAL_TRAIT_ONNX_EDITOR_TEST_SUPPORTED TRUE)
```

What this FindOnnx.cmake script can do is to create a Platform Abstract Layer layer.
You can look at the Atom RHI Gem declaration of the 3rdParty RenderDoc target as an example of declaring a platform specific target (https://github.com/o3de/o3de/tree/development/Gems/Atom/RHI/3rdParty).
You can also reach out to me if you need help set it up

Reply:
Changed